api/src/main/java/com/velocitypowered/api/network/ProtocolVersion.java

```diff
@@ -0,0 +1,178 @@
+    MINECRAFT_1_12_1(338, "1.12.1"),
+    MINECRAFT_1_12_2(340, "1.12.2"),
+    MINECRAFT_1_13(393, "1.13"),
+    MINECRAFT_1_13_1(401, "1.13.1"),
+    MINECRAFT_1_13_2(404, "1.13.2");
+
+    private final int protocol;
+    private final String name;
+
+    /**
+     * Represents the lowest supported version
+     */
+    public static final ProtocolVersion MINIMUM_VERSION = MINECRAFT_1_8;
+    /**
+     * Represents the highest supported version
+     */
+    public static final ProtocolVersion MAXIMUM_VERSION = MINECRAFT_1_13_2;
+
+    /**
+     * The user-friendly representation of the lowest and highest supported versions
+     */
+    public static final String SUPPORTED_VERSION_STRING = String.format("%s-%s", MINIMUM_VERSION, MAXIMUM_VERSION);
+
+    /**
+     * A map linking the protocol version number to its {@link ProtocolVersion} representation
+     */
+    public static final ImmutableMap<Integer, ProtocolVersion> ID_TO_PROTOCOL_CONSTANT;
+
+    static {
+        Map<Integer, ProtocolVersion> versions = new HashMap<>();
+        for (ProtocolVersion version : values()) {
+            versions.put(version.protocol, version);
+        }
+
+        ID_TO_PROTOCOL_CONSTANT = ImmutableMap.copyOf(versions);
+    }
+
+    ProtocolVersion(int protocol, String name) {
+        this.protocol = protocol;
+        this.name = name;
+    }
+
+    /**
+     * Returns the protocol as an int
+     *
+     * @return the protocol version
+     */
+    public int getProtocol() {
+        return protocol;
+    }
+
+    /**
+     * Returns the user-friendly name for this protocol
+     *
+     * @return the protocol name
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Gets the {@link ProtocolVersion} for the given protocol
+     *
+     * @param protocol the protocol as an int
+     * @return the protocol version
+     */
+    public static ProtocolVersion getProtocolVersion(int protocol) {
+        return ID_TO_PROTOCOL_CONSTANT.getOrDefault(protocol, UNKNOWN);
+    }
+
+    /**
+     * Returns whether the protocol is supported
+     *
+     * @param protocol the protocol as an int
+     * @return if the protocol supported
+     */
+    public static boolean isSupported(int protocol) {
+        ProtocolVersion version = ID_TO_PROTOCOL_CONSTANT.get(protocol);
+
+        return version != null && !version.isUnknown();
+    }
+
+    /**
+     * Returns whether the {@link ProtocolVersion} is supported
+     *
+     * @param version the protocol version
+     * @return if the protocol supported
+     */
+    public static boolean isSupported(ProtocolVersion version) {
+        return version != null && !version.isUnknown();
+    }
+
+    /**
+     * Returns whether this {@link ProtocolVersion} is greater than the {@link ProtocolVersion} provided
+     *
+     * @param version the protocol version
+     * @return if the protocol is greater
+     */
+    public boolean isGreater(ProtocolVersion version) {
```

Comment:
How about replacing all of these comparison methods with a [compare logic similar to Integer](https://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#compare(int,%20int))?

Reply:
These are already ordered, so we can simply use the `compareTo()` implementation baked into `Enum`.